HW 1 - Palindrome

hickamt opened the pull request:
Sorry, I didn't realize that when pulling from CLI for main that it would close the pull request.

# Palindrome
1. From the original file I have abstracted functionality into atomic functions
2. The event listener uses higher order functions including the pipe() function which stacks the call flow

## Why changes were made
1. Read the textbook 'learning react' which talks about functional programming and higher order functions and wanted to apply these principles
2. After reading about pipe() and compose() I wanted to try and implement the functionality of Palindrome using this type of higher order functionality. I have modified my function calls to be as descriptive as possible

## What I would like for feedback
1. Are the functions self describing and correct
2. Is the program easy to follow/read using the pipe() stack call
3. Are there other improvements that could be made

linh224 reviewed (commented):
Hello, Todd. 
I can no longer comment on.js and.html files. I expected to see three files altered, but now I only see one. So I'll simply leave it here.
- Palindrome.html file: Do we need to use **bg-primary** on line 12 to get the blue like the professor's sample? 
- In the file palindrome.js: I believe you are adhering to the Airbnb JavaScript Style Guide to a T. Your coding taught me a lot. Actually, before I saw your code, I had not considered creating constants p_success, p_failure, and p_error like you. It is quite beneficial to me. Thank you very much!

JoshuaWachana commented:
Hello Todd,

I also had trouble commenting on the files. As you hinted it seems like you may have accidently closed the earlier ones. So I will just make some comments here.

Overall, great job getting the assignment done. Most of it looks pretty good, but I do have a few comments to make.

First off, it looks like the background color of Ex02 is incorrect. I believe that the screenshots showed a blue background, whereas yours has a dark/black background. This can be fixed by changing `<body class="bg-dark">` on line 12 of your palindrome.html to `<body class="bg-primary">` .

Next, I noticed a small issue in terms of mobile responsiveness. I think going forward you should make sure to test out your application on smaller screen sizes. The image below shows what I saw on my end when I tested it out on a small device.

![thmobilesize](https://github.com/hickamt/CS464-Front-End-HW/assets/38127011/21d2c39d-9a7d-4ecf-a26f-a80106154a37)

Also, on line 49 of palindrome.js you use a template literal to build out the p element `<p id=${id} class="text-${color} fs-${fontSize} fw-${fontWeight}">${message}</p>;` . Even though this is an option, I feel like this will be harder to maintain and it is a bit hard to read. I feels a bit manual and like hardcoded html. It may be simpler to just build out elements using methos like append() even though it may use more lines of code.

Good work and have a nice day!

hickamt commented:
As noted in the revision I have changed/modified the following:

1. changed the body background color to 'primary'

2. I have included the necessary classes in the HTML to be responsive for mobile

3. I have re-written the reusable paragraph template literal in a way that is easier to read

4. The constants used for success, failure, and error messages have been reduced and the 'createHtmlElement()' now contains default de-structured arguments

Thank you for the advice/review of the program. I have finally learned to create a new branch for each of the exercises and will leave them open until reviewed for HW-02.
